src/frontend/src/optimizer/logical_optimization.rs

```diff
@@ -426,6 +426,13 @@ impl LogicalOptimizer {
         explain_trace: bool,
         ctx: &OptimizerContextRef,
     ) -> Result<PlanRef> {
+        // In order to unnest a table function, we need to convert it into a `project_set` first.
+        // However, the table function is expected to be converted into a project set, so it must be applied.
```

Comment:
What does this new comment mean?

Reply:
This rule is also to make table function work in streaming? 🥵

Reply:
BTW, we also discussed whether to abandon the table function operator https://github.com/risingwavelabs/risingwave/pull/11422#pullrequestreview-1564719067

Not sure whether we can/need to put `TABLE_FUNCTION_TO_PROJECT_SET` outside `subquery_unnesting`

Reply:
I mean this rule it necessary right now. Maybe I should move this rule outside🤔
https://github.com/risingwavelabs/risingwave/blob/4ddcee46cf50b0afc29f979f6816b89d12b63a44/src/frontend/src/optimizer/plan_node/logical_table_function.rs#L124-L140

Reply:
> This rule is also to make table function work in streaming? 🥵

Yes